nixos/modules/services/x11/clight.nix

```diff
@@ -0,0 +1,112 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+let
+  cfg = config.services.clight;
+
+  toConf = { isAttr ? false }@args: v:
+    if builtins.isFloat v then toString v
+    else if isList v then "[ " + concatMapStringsSep ", " (toConf {}) v + " ]"
+    else if isAttr && isString v then toString v
+    else generators.toPretty {} v;
+
+  clightConf = {
+    inherit (config.location) latitude longitude;
+    gamma_temp = with cfg.temperature; [ day night ];
```

Comment:
You can set these values in the `config` section of the module:

```nix
{
  config = {
    services.clight.settings = {
      location.latitude = mkDefault config.location.latitude;
      location.longitude = mkDefault config.location.longitude;
      gamma_temp = mkDefault [ cfg.temperature.day cfg.temperature.night ];
    };
  };
}
```

(`mkDefault` such that the user can override the values directly through `.settings` if they want to)